neovim/api/buffers.py

```diff
@@ -0,0 +1,53 @@
+"""API for working with NVim buffers.
+
+Conforms to python-buffers, which reads:
+
+  vim.buffers                        *python-buffers*
+    A mapping object providing access to the list of vim buffers.  The
+    object supports the following operations: >
+        :py b = vim.buffers[i]    # Indexing (read-only)
+        :py b in vim.buffers    # Membership test
+        :py n = len(vim.buffers)    # Number of elements
+        :py for b in vim.buffers:    # Iterating over buffer list
+
+The key for indexing into vim.buffers[] is the buffer number as described in
+*python-buffer*.
+"""
+
+import functools
+
+__all__ = ('Buffers')
+
+
+class Buffers(object):
```

Comment:
should be in `api/nvim.py` as it is tightly coupled to the nvim class. compare `nvim.current`